### PR TITLE
Fixed Scroll up and down to zoom in and out respectively

### LINKED ISSUE #828: Scroll up to Zoom in and Scroll down to Zoom out
**Describe the bug**
In the simulator when you scroll up it should zoom in but it is happening the opposite.

**To Reproduce**
1. Go to https://circuitverse.org/simulator
2. Scroll Up

**Expected behavior**
When you scroll up, it should zoom in and not zoom out

**Screenshots**

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #828 Scroll up to Zoom in and Scroll down to Zoom out

#### Describe the changes you have made in this pr - I have reversed the actions performed by the Scrolling Up and down . 

 - [x] All Checks Passed Successfully

### CODE REVIEW

**Comment by Shivansh2407** [comment]
> Please update the PR title to add a very concise description of the changes introduces
> also if possible , please ad screenshots(maybe a gif)

Sure @thiefie

**Comment by Shivansh2407** [comment]
Please Review and  Merge this PR @JoshHeng @thiefie @satu0king

**Comment by Ananthavijay** [comment]
@thiefie Does this works? I tried this is in gitpod but there was no change. If it works could u please check that control+scroll has an inverted zoom? As @shubhankarsharma00 said some may prefer the inverted zoom.

File: public/js/embedListeners.js [hunk 1]
```diff
@@ -175,21 +175,21 @@ function startListeners() {
...
         var deltaY = event.wheelDelta ? event.wheelDelta : -event.detail;
-        var scrolledUp = deltaY < 0;
-        var scrolledDown = deltaY > 0;
+        var scrolledDown = deltaY < 0;
+        var scrolledUp = deltaY > 0;
 
         if (event.ctrlKey) {
-            if (scrolledUp && globalScope.scale > 0.5 * DPR) {
+            if (scrolledDown && globalScope.scale > 0.5 * DPR) {
                 changeScale(-.1 * DPR);
```
Comment (thiefie) [inline public/js/embedListeners.js]: You shouldn't have changes this as you have already inverted the values above.
Please revert back to `scrolledUp` only.

File: public/js/embedListeners.js [hunk 2]
```diff
@@ -175,21 +175,21 @@ function startListeners() {
...
         event.preventDefault()
         var deltaY = event.wheelDelta ? event.wheelDelta : -event.detail;
-        var scrolledUp = deltaY < 0;
-        var scrolledDown = deltaY > 0;
+        var scrolledDown = deltaY < 0;
+        var scrolledUp = deltaY > 0;
 
         if (event.ctrlKey) {
-            if (scrolledUp && globalScope.scale > 0.5 * DPR) {
+            if (scrolledDown && globalScope.scale > 0.5 * DPR) {
```
Comment (thiefie) [inline public/js/embedListeners.js]: You shouldn't have changes this as you have already inverted the values above.
Please revert back to `scrolledUp` only.

File: public/js/embedListeners.js [hunk 3]
```diff
@@ -175,21 +175,21 @@ function startListeners() {
...
                 changeScale(-.1 * DPR);
             }
-            if (scrolledDown && globalScope.scale < 4 * DPR) {
+            if (scrolledUp && globalScope.scale < 4 * DPR) {
```
Comment (thiefie) [inline public/js/embedListeners.js]: You shouldn't have changes this as you have already inverted the values above.
Please revert back to `scrolledDown` only.

File: public/js/embedListeners.js [hunk 4]
```diff
@@ -175,21 +175,21 @@ function startListeners() {
...
                 changeScale(.1 * DPR);
             }
         } else {
-            if (scrolledUp && globalScope.scale < 4 * DPR) {
+            if (scrolledDown && globalScope.scale < 4 * DPR) {
                 changeScale(.1 * DPR);
```
Comment (thiefie) [inline public/js/embedListeners.js]: You shouldn't have changes this as you have already inverted the values above.
Please revert back to `scrolledUp` only.

File: public/js/embedListeners.js [hunk 5]
```diff
@@ -175,21 +175,21 @@ function startListeners() {
...
             }
-            if (scrolledDown && globalScope.scale > 0.5 * DPR) {
+            if (scrolledUp && globalScope.scale > 0.5 * DPR) {
```
Comment (thiefie) [inline public/js/embedListeners.js]: Same here.

File: public/js/listeners.js [hunk 6]
```diff
@@ -221,25 +221,24 @@ function startListeners() {
...
         event.preventDefault()
         var deltaY = event.wheelDelta ? event.wheelDelta : -event.detail;
-        var scrolledUp = deltaY < 0;
-        var scrolledDown = deltaY > 0;
+        var scrolledDown = deltaY < 0;
+        var scrolledUp = deltaY > 0;
 
         if (event.ctrlKey) {
-            if (scrolledUp && globalScope.scale > 0.5 * DPR) {
+            if (scrolledDown && globalScope.scale > 0.5 * DPR) {
```
Comment (thiefie) [inline public/js/listeners.js]: Please do similar changes in this file also

**Comment by thiefie** [comment]
@Shivansh2407 @Ananthavijay The changes don't work as you inverted the values of each event and then also swapped the variables which caused no changes.
The inverted zoom will also be inverted

**Comment by Shivansh2407** [comment]
Ok Cool. Will work on the requested changes.

**Comment by Shivansh2407** [comment]
@thiefie Please check it now and let me know. Also please guide me how to view it on gitpod?

**Comment by Ananthavijay** [comment]
@Shivansh2407 I will check it dont worry

**Comment by thiefie** [comment]
Gitpod files are configured by default so you can just go to `https://gitpod.io/#<URL-OF-PULL-REQUEST>` , and you could see a version of the code hosted on the cloud. And you could also test it there ,

**Comment by Ananthavijay** [comment]
Please squash and change the commit message. Thanks for your contribution!

**Comment by Shivansh2407** [comment]
Thank You @Ananthavijay .

**Comment by thiefie** [comment]
For any help these maybe useful:
1) Commit Messages: https://chris.beams.io/posts/git-commit/
2) Squashing Commits: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

@shubhankarsharma00 @tachyons can this be merged?

**Comment by Shivansh2407** [comment]
Actually I have the commits in a branch patch1 of the forked repo so can you guide me how to continue to squash and commit @thiefie @Ananthavijay

**Comment by thiefie** [comment]
You need to first clone the forked repo locally (if you already have not
done so) using `git clone <URL-OF-FORK>`
Then checkout that branch using `git checkout <BRANCH-NAME>`
Then maybe follow the steps suggested in the site I linked above , if you
get stuck feel free to ask here or in the slack channel

**Comment by Shivansh2407** [comment]
I think so I did it ! Thank you @thiefie and @Ananthavijay

**Comment by Shivansh2407** [comment]
@tachyons and @Ananthavijay could you please review and merge this PR.

**Comment by Shivansh2407** [comment]
@thiefie and @tachyons please approve and merge this PR

**Comment by tachyons** [comment]
Simulator changes need to be approved by @satuking

**Comment by Shivansh2407** [comment]
@tachyons and @satu0king Please Review and merge 👍

**Comment by Shivansh2407** [comment]
@satu0king Please review this 👍

**Comment by Shivansh2407** [comment]
@satu0king Please review this and merge 👍

**Comment by Shivansh2407** [comment]
@satu0king Please review and merge 👍

**Comment by Shivansh2407** [comment]
@satu0king please review and merge 👍

**Comment by satu0king** [comment]
@Shivansh2407 Please refer comment on original issue
https://github.com/CircuitVerse/CircuitVerse/issues/828#issuecomment-605598078

**Pull request closed without merging**